src/compiler/types.ts

```diff
@@ -3663,16 +3663,27 @@ namespace ts {
         errors: Diagnostic[];
         wildcardDirectories?: MapLike<WatchDirectoryFlags>;
         compileOnSave?: boolean;
+        configFileSpecs?: ConfigFileSpecs;
     }
 
     export const enum WatchDirectoryFlags {
         None = 0,
         Recursive = 1 << 0,
     }
 
+    export interface ConfigFileSpecs {
+        filesSpecs: string[];
+        includeSpecs: string[];
+        excludeSpecs: string[];
+        validatedIncludeSpecs: string[];
+        validatedExcludeSpecs: string[];
+        wildcardDirectories: MapLike<WatchDirectoryFlags>;

```

Comment:
I want to keep this MapLike just the way json has to ensure that the conversion is done only on need basis